hera_cal/reflections.py

```diff
@@ -267,25 +239,445 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+                                   edgecut_low=edgecut_low, edgecut_hi=edgecut_hi)
+            model_fft = np.true_divide(model_fft, win, where=~np.isclose(win, 0.0))
+
+            # subtract from data
+            self.pcomp_model_fft[k] = model_fft
+            self.data_pmodel_resid[k] = data[k] - model_fft
+
+    def interp_u(self, u, times, full_times=None, uflags=None, keys=None, overwrite=False, verbose=True,
+                 mode='gpr', gp_len=600, gp_nl=0.1, optimizer=None):
+        """
+        Interpolate u modes along time, inserts into self.umode_interp
+
+        Args:
+            u : DataContainer
+                u-mode container to interpolate
+            times : 1D array
+                Holds time_array of input u modes.
+            full_times : 1D array
+                time_array to interpolate onto. Default is times.
+            uflags : DataContainer
+                Object to pull target u flags from. Default is None.
+            keys : list of tuples
+                List of baseline-pol DataContainer tuples to operate on.
+            overwrite : bool
+                If True, overwrite output data if it exists.
+            verbose : bool
+                If True, report feedback to stdout.
+            mode : str
+                Interpolation mode. Options=['gpr']
+            gp_len : length-scale of GPR in units of times
+            gp_nl : GPR noise-level in units of input u.
+        """
+        if not hasattr(self, 'umode_interp'):
+            self.umode_interp = DataContainer({})
+
+        if uflags is None:
+            uflags = self.uflags
+
+        if full_times is None:
+            full_times = times
+
+        # get keys
+        if keys is None:
+            keys = u.keys()
+
+        # setup X predict
+        Xmean = np.median(times)
+        Xpredict = full_times[:, None] - Xmean
+
+        # iterate over keys
+        for k in keys:
+            # check overwrite
+            if k in self.umode_interp and not overwrite:
+                echo("{} in umode_interp and overwrite == False, skipping...".format(k), verbose=verbose)
+                continue
+
+            if mode == 'gpr':
+                # setup GP kernel
+                kernel = 1**2 * gp.kernels.RBF(length_scale=gp_len) + gp.kernels.WhiteKernel(noise_level=gp_nl)
+                GP = gp.GaussianProcessRegressor(kernel=kernel, optimizer=optimizer, normalize_y=True)
+
+                # setup regression data: get unflagged data
+                select = ~np.max(uflags[k], axis=1)
+                X = times[select, None] - Xmean
+                y = u[k][select, :]
+
+                # fit gp and predict
+                GP.fit(X, y.real)
+                ypredict_real = GP.predict(Xpredict)
+                GP.fit(X, y.imag)
+                ypredict_imag = GP.predict(Xpredict)
+
+                # append
+                self.umode_interp[k] = ypredict_real.astype(np.complex) + 1j * ypredict_imag
```

Comment:
The `.astype(np.complex)` is unnecessary. The sum will handle the type conversion.